### PR TITLE
Refactor multiples logcollector test

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|-------------|
|      [3218](https://github.com/wazuh/wazuh-qa/issues/3218)   |

## Description

This PR aims to fix the following tests reported as unstable:
 - `test_logcollector/test_configuration/test_basic_configuration_age.py`
 - `test_logcollector/test_configuration/test_basic_configuration_query.py`
 - `test_logcollector/test_command_monitoring/test_command_execution_freq.py `
 - `test_logcollector/test_configuration/test_basic_configuration_only_future_events.py`
 - `test_logcollector/test_command_monitoring/test_command_execution_dbg.py`
 
A full refactor of the test with the new QA structure has been carried on.

### Added

For `test_command_execution_freq` it has added a fixture that modifies the date format of the Windows OS to use the `TimeMachine.travel_to_future` function. 

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @fedepacher  (Developer)  |           |  :green_circle::green_circle::green_circle:  | :green_circle::green_circle::green_circle: |   Windows/Linux      |         | Nothing to highlight |
| @Rebits  (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

Jenkins Launches:
- [Launch 1](https://ci.wazuh.info/job/Test_integration/32926/console)
- [Launch 2](https://ci.wazuh.info/job/Test_integration/32947/console)
- [Launch 3](https://ci.wazuh.info/job/Test_integration/32948/console)

### CODE REVIEW

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def make_logcollector_callback(pattern, prefix=LOG_COLLECTOR_PREFIX, escape=Fals
 
 
 def check_logcollector_event(file_monitor=None, callback='', error_message=None, update_position=True,
-                             timeout=T_30, prefix=LOG_COLLECTOR_PREFIX, accum_results=1, file_to_monitor=LOG_FILE_PATH,
+                             timeout=T_60, prefix=LOG_COLLECTOR_PREFIX, accum_results=1, file_to_monitor=LOG_FILE_PATH,
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply deps/wazuh_testing/wazuh_testing/modules/logcollector/event_monitor.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/1eb09285008fbcbeac422a85949d5cea08d3cb77)

File: deps/wazuh_testing/wazuh_testing/tools/file.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def read_yaml(file_path):
     Returns:
        dict: Yaml structure.
     """
-    with open(file_path) as f:
+    with open(file_path, encoding='utf-8') as f:
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply deps/wazuh_testing/wazuh_testing/tools/file.py:50]: It has launched all the tests that use this function and have passed successfully.

**Review by jmv74211 — CHANGES_REQUESTED** [review]
@fedepacher this test `test_logcollector/test_configuration/test_basic_configuration_age.py` has failed in windows. Take a look at the [Jenkins build](https://ci.wazuh.info/job/Test_integration/35925/artifact/windows_agent_tests_Test_integration_B35925_20230124155225.txt)

```
================================== FAILURES ===================================
______________________ test_configuration_age[syslog_3s] ______________________
test_logcollector\test_configuration\test_basic_configuration_age.py:186: in test_configuration_age
    check_configuration_age_valid(metadata)
test_logcollector\test_configuration\test_basic_configuration_age.py:123: in check_configuration_age_valid
    evm.check_analyzing_file(file=metadata['location'], error_message=error_message, prefix=prefix)
C:\Python37\lib\site-packages\wazuh_testing-4.4.0-py3.7.egg\wazuh_testing\modules\logcollector\event_monitor.py:68: in check_analyzing_file
    error_message=error_message, prefix=prefix)
C:\Python37\lib\site-packages\wazuh_testing-4.4.0-py3.7.egg\wazuh_testing\modules\logcollector\event_monitor.py:54: in check_logcollector_event
    callback=make_logcollector_callback(callback, prefix, escape), error_message=error_message)
C:\Python37\lib\site-packages\wazuh_testing-4.4.0-py3.7.egg\wazuh_testing\tools\monitoring.py:203: in start
    error_message=error_message).result()
C:\Python37\lib\site-packages\wazuh_testing-4.4.0-py3.7.egg\wazuh_testing\tools\monitoring.py:469: in start
    raise TimeoutError(error_message)
E   TimeoutError: The expected analyzing file log has not been produced
---------------------------- Captured stdout call -----------------------------


The Wazuh service was stopped successfully.



The Wazuh service is starting.

The Wazuh service was started successfully.



---------------------------- Captured stderr call -----------------------------
2023-01-24 15:17:40,508 - wazuh_testing - ERROR - The expected analyzing file log has not been produced
2023-01-24 15:17:40,508 - wazuh_testing - ERROR - Results accumulated: 0
2023-01-24 15:17:40,508 - wazuh_testing - ERROR - Results expected: 1
------------------------------ Captured log call ------------------------------
ERROR    wazuh_testing:monitoring.py:465 The expected analyzing file log has not been produced
ERROR    wazuh_testing:monitoring.py:466 Results accumulated: 0
ERROR    wazuh_testing:monitoring.py:468 Results expected: 1
- generated html file: file://C:\tmp\Test_integration_B35925_20230124155225\report.html -
=========================== short test summary info ===========================
FAILED test_logcollector/test_configuration/test_basic_configuration_age.py::test_configuration_age[syslog_3s]
= 1 failed, 428 passed, 569 skipped, 180 deselected, 39 xfailed, 5 xpassed in 1926.91s (0:32:06) =
```

**Comment by davidjiglesias** [comment]
Due to the migration of the IT to the devel teams, and the instability of the tests included in this PR, we have sadly decided to close the PR without including the changes, as they might bring even more issues to the table.

**Pull request closed without merging**